modules/ding_provider/ding_provider.module

```diff
@@ -605,6 +598,85 @@ function ding_provider_get_pseudo_entity($ding_entity_id) {
   return $ting_entity;
 }
 
+/**
+ * Use this to try and create pseudo entities when materials are not searchable
+ * in the well with current agencyid.
+ * Note: Use this instead of ding_provider_get_pseudo_entity(), which is only
+ * kept for legacy support. This function will try
+ * ding_provider_get_pseudo_entity() if it's implemented in the provider.
+ *
+  * @param array $ding_entity_ids
+ *   Array of Ding entity ids (data-well PIDs) to try and find pseudo entities
+ *   for.
+ *
+ * @return array
+ *   Array of pseudo-entities keyed by ding_entity_id. If the function couldn't
+ *   find a pseudo entity, the corresponding entry will be NULL.
+ */
+function ding_provider_get_pseudo_entities($ding_entity_ids) {
```

Comment:
The input parameter `$ding_entity_ids` should be either type-casted as array, or a check performed to transform single value to array-like, placed in the very beginning of the function. Hence, string can also be iterated as arrays, which is not the case here.